sam2lca/sam_pysam.py

```diff
@@ -21,7 +21,10 @@ def __init__(self, al_file):
         alignment = pysam.AlignmentFile(al_file, mode[filetype])
         self.al_file = al_file
         self.mode = mode[filetype]
-        self.refs = alignment.references
+        present_refs = set()
+        for read in alignment:
+            set.add(read.reference_name)
+        self.refs = tuple(present_refs)
```

Comment:
Going through all the reads is a bit inneficient, but lucklily there is the [get_index_statistics](https://pysam.readthedocs.io/en/latest/api.html#pysam.AlignmentFile.get_index_statistics) method 😃 
```suggestion
        present_refs = set()
        for ref_stat in alignment.get_index_statistic():
            refname = ref_stat[0]
            nb_mapped_reads = ref_stat[1]
            if nb_mapped_reads > 0:
                present_refs.add(refname)
        self.refs = tuple(present_refs)
```